lib/apiservers/engine/backends/container.go

```diff
@@ -836,7 +836,40 @@ func (c *Container) ContainerInspect(name string, size bool, version version.Ver
 // ContainerLogs hooks up a container's stdout and stderr streams
 // configured with the given struct.
 func (c *Container) ContainerLogs(name string, config *backend.ContainerLogsConfig, started chan struct{}) error {
-	return fmt.Errorf("%s does not implement container.ContainerLogs", ProductName())
+	//FIXME:  Doug, remove following 3 lines once integration is added
+	if true {
+		return fmt.Errorf("%s does not implement container.ContainerLogs", ProductName())
+	}
+	defer trace.End(trace.Begin("ContainerAttach"))
+
+	// Look up the container name in the metadata cache to get long ID
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+	}
+	name = vc.ContainerID
+
+	tailLines, since, follow, err := c.validateContainerLogsConfig(vc, config)
+	if err != nil {
+		return err
+	}
+
+	// Outstream modification (from Docker's code) so the stream is streamed with the
+	// necessary headers that the CLI expects.  This is Docker's scheme.
+	wf := ioutils.NewWriteFlusher(config.OutStream)
+	defer wf.Close()
+	close(started)
+	wf.Flush()
+
+	var outStream io.Writer = wf
+	if !vc.Config.Tty {
+		outStream = stdcopy.NewStdWriter(outStream, stdcopy.Stdout)
+	}
+
+	// Make a call to our proxy to handle the remoting
+	err = c.containerProxy.StreamContainerLogs(name, outStream, started, config.Timestamps, follow, since, tailLines)
```

Comment:
It wasn't always this way.  At one point, we were using vc.ContainerID, but a few PR back, there was a suggestion that we overwrite name for stylistic reason.  I didn't care so I started adopting it, but if it taking away functionality, then we should stop doing it.